Tests/PHPUnit/Components/StatusConverter/PaymentStatusConverterTest.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace MollieShopware\Tests\Components\StatusConverter;
+
+use MollieShopware\Components\Constants\PaymentStatus;
+use MollieShopware\Components\StatusConverter\PaymentStatusConverter;
+use PHPUnit\Framework\TestCase;
+use Shopware\Models\Order\Status;
+
+
+class PaymentStatusConverterTest extends TestCase
+{
+
+    /**
+     *
+     */
+    public const CUSTOM_STATUS_AUTHORIZED = 99;
+
+
+    /**
+     * This test verifies that our Mollie Payment status is
+     * correctly converted into the matching Shopware Payment Status
+     * Here are overviews about the status changes in Mollie:
+     * - https://docs.mollie.com/payments/status-changes
+     * - https://docs.mollie.com/orders/status-changes
+     *
+     * @dataProvider getStatusMappings
+     * @covers       \MollieShopware\Components\StatusConverter\PaymentStatusConverter::getShopwarePaymentStatus
+     *
+     * @param $status
+     * @param $expectedStatus
+     * @throws \MollieShopware\Exceptions\PaymentStatusNotFoundException
+     */
+    public function testStatusMapping($status, $expectedStatus)
+    {
+        $converter = new PaymentStatusConverter(self::CUSTOM_STATUS_AUTHORIZED);
+
+        $result = $converter->getShopwarePaymentStatus($status);
+
+        # if we have no target state, that means we
+        # just ignore that state
+        $expectedIgnoreState = ($expectedStatus === null);
+
+        $this->assertEquals($expectedStatus, $result->getTargetStatus());
+        $this->assertEquals($expectedIgnoreState, $result->isIgnoreState());
+    }
+
+    /**
+     * @return array[]
+     */
+    public function getStatusMappings()
```

Comment:
reviewers, please verify our source and target status..
null means it will do nothing
if you have any concerns just let me know, thank you